### ISSUE TITLE
restore with polymorphic relation fails

### ISSUE THREAD

**Issue by hnishide** [rubysherpas/paranoia#174]
When restoring, it search for dependent table's records by the foreign key composed of parent's class name as it does when dealing with simple non-polymorphic relation, eventually fails to find the column and crush.

**Comment by hamidumi** [comment]
I overridden method restore_associated_records. Now recursive restore can restore polymorphic has one associations. I don't know how about has many associations, but it can be done in the same manner. I used rails4 branch.
My code:

```
module Paranoia
  def restore_associated_records
    destroyed_associations = self.class.reflect_on_all_associations.select do |association|
      association.options[:dependent] == :destroy
    end

    destroyed_associations.each do |association|
      association_data = send(association.name)

      unless association_data.nil?
        if association_data.paranoid?
          if association.collection?
            association_data.only_deleted.each { |record| record.restore(:recursive => true) }
          else
            association_data.restore(:recursive => true)
          end
        end
      end

      if association_data.nil? && association.macro.to_s == "has_one"
        association_class_name = association.class_name.present? ? association.class_name : association.name.to_s.camelize
        association_foreign_key = association.foreign_key.present? ? association.foreign_key : "#{self.class.name.to_s.underscore}_id"
        binding.pry
        if association.type
          association_polymorphic_type = association.type
          Object.const_get(association_class_name).only_deleted.where(association_polymorphic_type => self.class.name.to_s, association_foreign_key => self.id).first.try(:restore, recursive: true)
        else
          Object.const_get(association_class_name).only_deleted.where(association_foreign_key, self.id).first.try(:restore, recursive: true)
        end
      end
    end

    clear_association_cache if destroyed_associations.present?
  end
end
```

**Comment by jhawthorn** [comment]
This should be fixed in `2.0.5`

**Comment by hnishide** [comment]
Thanks!
